cloud-core/src/main/java/cloud/commandframework/arguments/standard/DurationArgument.java

```diff
@@ -0,0 +1,279 @@
+     * @param <C>  Command sender type
+     * @return Created builder
+     */
+    public static <C> @NonNull Builder<C> newBuilder(final @NonNull String name) {
+        return new Builder<>(name);
+    }
+
+    /**
+     * Create a new required command component
+     *
+     * @param name Component name
+     * @param <C>  Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> of(final @NonNull String name) {
+        return DurationArgument.<C>newBuilder(name).asRequired().build();
+    }
+
+    /**
+     * Create a new optional command component
+     *
+     * @param name Component name
+     * @param <C>  Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> optional(final @NonNull String name) {
+        return DurationArgument.<C>newBuilder(name).asOptional().build();
+    }
+
+    /**
+     * Create a new required command component with a default value
+     *
+     * @param name            Component name
+     * @param defaultDuration Default duration
+     * @param <C>             Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> optional(
+            final @NonNull String name,
+            final @NonNull String defaultDuration
+    ) {
+        return DurationArgument.<C>newBuilder(name).asOptionalWithDefault(defaultDuration).build();
+    }
+
+
+    public static final class Builder<C> extends CommandArgument.Builder<C, Duration> {
+
+        private Builder(final @NonNull String name) {
+            super(Duration.class, name);
+        }
+
+        /**
+         * Builder a new boolean component
+         *
+         * @return Constructed component
+         */
+        @Override
+        public @NonNull DurationArgument<C> build() {
+            return new DurationArgument<>(this.isRequired(), this.getName(), this.getDefaultValue(),
+                    this.getSuggestionsProvider(), this.getDefaultDescription()
+            );
+        }
+
+    }
+
+
+    public static final class DurationParser<C> implements ArgumentParser<C, Duration> {
+
+        @Override
+        public @NonNull ArgumentParseResult<Duration> parse(
+                final @NonNull CommandContext<C> commandContext,
+                final @NonNull Queue<String> inputQueue
+        ) {
+            final String input = inputQueue.peek();
+            if (input == null) {
+                return ArgumentParseResult.failure(new NoInputProvidedException(
+                        DurationArgument.DurationParseException.class,
+                        commandContext
+                ));
+            }
+            inputQueue.remove();
+
+            /*
+             * Matches durations in the format of: "2d15h7m12s"
+             */
+            final Matcher matcher = Pattern.compile("(([1-9][0-9]+|[1-9])[dhms])").matcher(input);
```

Comment:
should this pattern be compiled and stored as a constant so that we don’t recompile a pattern per-parse?